### ISSUE TITLE
Browserify standalone + expose

### ISSUE THREAD

**Issue by drkibitz** [GoodBoyDigital/pixi.js#1574]
Doesn't look like using the `standalone = 'PIXI'` and `expose: 'pixi.js` work together. Using `standalone` prevents anything from being exposed. It's either one or the other.

I'm not sure of what to do here. I guess a proposal would be defining `global.PIXI` in `src/index.js`. This would allow using the same bundle define `'pixi.js'` as an external and define the PIXI global.

Otherwise, it would be 2 bundles, one standalone, and one as an external.

**Comment by drkibitz** [comment]
Pretty sure this is because using `standalone` wraps the bundle in a UMD closure, and prevents the bundle's `require` function from being exposed, regardless of `externalRequireName` option.

**Comment by englercj** [comment]
I see, I wish I could remember the use-case for wanting to expose pixi.js.

If you have a bundle next to pixi you can just use the global

**Comment by GoodBoyDigital** [comment]
think it may have been for plugins?

**Comment by englercj** [comment]
I thought so too, so external plugins could be bundled separately and still use `require('pixi.js')`. I wonder then which of the following options we should do:

1) Use expose in browserify, and remove standalone. We then include an export ourselves that UMD exports pixi in our `src/index.js` file.

OR

2) We continue using standalone, drop expose and make external plugins use the globally exported PIXI. Not sure if this will work for all environments (say requirejs) though...

**Comment by GoodBoyDigital** [comment]
This one is definitely a problem that im not to sure how to solve! Happy to take advice as i'm sure there is a solution that @englercj  and @drkibitz can come up with.

For me the goal is to be able to make a plugin that works with both browserfy and requirejs (as we are currently using that for our projects)

https://github.com/GoodBoyDigital/pixi.js/tree/dev-flip is the perfect example of where to start with. I have no idea how to separate the flip folder into its own place and make it 'hook' into pixi to be a plugin. All ideas welcome!

**Comment by substack** [comment]
What about pointing the main field in package.json at the same entrypoint that browserify sees so that `require('pixi.js')` will work like usual first. Then build a separate dist target for other module formats as standalone.

**Comment by drkibitz** [comment]
@englercj Why did that change here? https://github.com/GoodBoyDigital/pixi.js/commit/d663658536325e0a2b870f1659a9c7ea394086e8

**Comment by drkibitz** [comment]
@englercj sorry my bad, that actually changed to be correct ;)

**Comment by drkibitz** [comment]
@substack see dev branch for the correct entrypoint.

**Comment by englercj** [comment]
@substack Sorry, should have specified that [dev][d] was the branch we are working on. There you can see we point main to the entry point `./src/index.js`. This makes normal browserify work fine.

What we want to accomplish is to allow users to use the module with standalone *and* to allow them to bundle their own plugins/apps using `--external pixi.js` and when they include our script and theirs it works.

Basically, we want UMD for requirejs and global exports and we also want people to externalize pixi from their browserify bundle.

Hopefully, that makes sense.

[d]: https://github.com/GoodBoyDigital/pixi.js/tree/dev

**Comment by englercj** [comment]
I removed the standalone so browserify only does expose. Then I added global and optional AMD exports to the index file, so those are exported manually.
